geth/api/backend_jail_test.go

```diff
@@ -765,3 +765,43 @@ func (s *BackendTestSuite) TestJailVMPersistence() {
 	s.T().Log(total)
 	require.InDelta(0.840003, total, 0.0000001)
 }
+
+func (s *BackendTestSuite) TestSendTransactionWithCallback() {
+	require := s.Require()
+
+	s.StartTestBackend(params.RopstenNetworkID)
+	defer s.StopTestBackend()
+
+	time.Sleep(TestConfig.Node.SyncSeconds * time.Second) // allow to sync
+
+	// log into account from which transactions will be sent
+	err := s.backend.AccountManager().SelectAccount(TestConfig.Account1.Address, TestConfig.Account1.Password)
+	require.NoError(err, "cannot select account: %v", TestConfig.Account1.Address)
+
+	jail := s.backend.JailManager()
+	jail.BaseJS(baseStatusJSCode)
+
+	result := jail.Parse(testChatID, `
+		_status_catalog['sendTestTx'] = function() {
```

Comment:
What is this `_status_catalog` variable?

Reply:
_status_catalog has am aware of is used by `status.js` and must be instantiated else it will blow up/error out on attempts to call any associated method of `status.js`. So this is actually necessary.
It is loaded by the `baseStatusJSCode`.